From f27b31eae52cc8ebed89360f7ef75a567daf0c64 Mon Sep 17 00:00:00 2001
From: Alpha Lin <alpha.lin@rock-chips.com>
Date: Fri, 13 Mar 2015 10:27:56 +0800
Subject: [PATCH] VPU: remove dev_set_name.

dev_set_name is prohibit in linux, and may casue system
warning dump.

Signed-off-by: Alpha Lin <alpha.lin@rock-chips.com>
---
 arch/arm/mach-rockchip/vcodec_service.c | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-rockchip/vcodec_service.c b/arch/arm/mach-rockchip/vcodec_service.c
index 91b055deae2a..01a1b46d1451 100644
--- a/arch/arm/mach-rockchip/vcodec_service.c
+++ b/arch/arm/mach-rockchip/vcodec_service.c
@@ -14,6 +14,8 @@
  *
  */
 
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
 #include <linux/clk.h>
 #include <linux/delay.h>
 #include <linux/init.h>
@@ -1951,7 +1953,7 @@ static int vcodec_subdev_probe(struct platform_device *pdev,
 
 	of_property_read_string(np, "name", (const char**)&name);
 	of_property_read_u32(np, "dev_mode", (u32*)&data->mode);
-	dev_set_name(dev, name);
+	/*dev_set_name(dev, name);*/
 
 	if (pservice->reg_base == 0) {
 		res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
@@ -2052,7 +2054,7 @@ static int vcodec_subdev_probe(struct platform_device *pdev,
 		goto err;
 	}
 
-	data->cls = class_create(THIS_MODULE, dev_name(dev));
+	data->cls = class_create(THIS_MODULE, name);
 
 	if (IS_ERR(data->cls)) {
 		ret = PTR_ERR(data->cls);
@@ -2061,7 +2063,7 @@ static int vcodec_subdev_probe(struct platform_device *pdev,
 	}
 
 	data->child_dev = device_create(data->cls, dev,
-		data->dev_t, NULL, dev_name(dev));
+		data->dev_t, NULL, name);
 
 	get_hw_info(data);
 
@@ -2069,18 +2071,18 @@ static int vcodec_subdev_probe(struct platform_device *pdev,
 
 	INIT_LIST_HEAD(&data->lnk_service);
 	list_add_tail(&data->lnk_service, &pservice->subdev_list);
-	
+
 #ifdef CONFIG_DEBUG_FS
 	data->debugfs_dir =
-		vcodec_debugfs_create_device_dir((char*)dev_name(dev), parent);
+		vcodec_debugfs_create_device_dir((char*)name, parent);
 	if (data->debugfs_dir == NULL)
-		vpu_err("create debugfs dir %s failed\n", dev_name(dev));
+		vpu_err("create debugfs dir %s failed\n", name);
 
 	data->debugfs_file_regs =
 		debugfs_create_file("regs", 0664,
 				    data->debugfs_dir, data,
 				    &debug_vcodec_fops);
-#endif	
+#endif
 	return 0;
 err:
 	if (data->irq_enc > 0)
-- 
2.35.3

